buzzmobile/process/bearing/bearing.py

```diff
@@ -40,13 +49,25 @@ def bearing(fix):
         lon1 = math.radians(bearings['last_fix'].longitude)
         lat2 = math.radians(fix.latitude)
         lon2 = math.radians(fix.longitude)
-        y = math.sin(lon2 - lon1) * math.cos(lat2)
-        x = math.cos(lat1)*math.sin(lat2) - math.sin(lat1)*math.cos(lat2)*math.cos(lon2 - lon1)
-        angle = math.atan2(y, x)
-        computedBearing = (angle + 2*math.pi) % 2*math.pi
-        bearings['med_filter'].add(computedBearing)
 
-        bearing_pub.publish(bearings['med_filter'].median())
+        #Calculates great-circle distance between positions
+        a = (math.pow(math.sin(lat2 - lat1), 2)
+             + math.cos(lat1) * math.cos(lat2)
+             * math.pow(math.sin((lon2 - lon1) / 2), 2))
+        c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+        distance = EARTH_RADIUS * c
+
+        if distance >= MIN_FIX_DISTANCE:
```

Comment:
So assuming the delta is not met means that the car is going too slow for bearings to be accurate I would say it should always be publishing a 0.